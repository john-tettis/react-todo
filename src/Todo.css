.fa-check-circle{
    float:right;
    display:inline;
    padding:.5rem;
    /* margin-right:1rem; */
}

.fa-check-circle:hover {
    color:rgb(69, 253, 69);
}

.todo-input{
	width: 90%;
	padding: 5px;
	border: none;
	box-sizing: border-box;
	height: 30px;
    background:transparent;
}

.todo-input:focus{
    outline:none;
}