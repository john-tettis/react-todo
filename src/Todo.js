import React,{useState} from 'react';
import {ListGroupItem,Form} from 'reactstrap'
import './Todo.css';


let Todo = ({text,id,deleteTodo,updateTodo})=>{
    let [value, setValue] = useState(text)
    let [valid, setValid] = useState(true)


    const handleSubmit=(e)=>{
        e.preventDefault();
        if(valid){
            updateTodo(id,value)
        }
    }
    const handleChange=(e)=>{
        setValue(e.target.value)
        setValid(v=>e.target.value!=='')
    }

    return(
            <ListGroupItem color={!valid? 'danger':''}key={id} onSubmit={handleSubmit}>
                <div className='row'>
                    <Form data-testid='form'className='col-10'>
                        <input  className='todo-input'value={value} onChange={handleChange} id={id}>
                        </input>  
                    </Form>
                    <div className='col-2'>
                        <i data-testid='delete' onClick={()=>deleteTodo(id)}className="far fa-check-circle fa-lg"></i>
                    </div>
                </div>
            </ListGroupItem>
    )

}

export default Todo