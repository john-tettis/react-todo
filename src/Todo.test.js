import { render, screen } from '@testing-library/react';
import Todo from './Todo';

test('renders ok', () => {
  render(<Todo/>);
});
test('snapshot is consistent', () => {
    let {asFragment}=render(<Todo />);
    expect(asFragment()).toMatchSnapshot()
  });
