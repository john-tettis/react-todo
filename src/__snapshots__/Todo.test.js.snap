// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot is consistent 1`] = `
<DocumentFragment>
  <li
    class="list-group-item"
  >
    <div
      class="row"
    >
      <form
        class="col-10"
        data-testid="form"
      >
        <input
          class="todo-input"
          value=""
        />
      </form>
      <div
        class="col-2"
      >
        <i
          class="far fa-check-circle fa-lg"
          data-testid="delete"
        />
      </div>
    </div>
  </li>
</DocumentFragment>
`;
